depclean-gradle-plugin/src/main/java/se/kth/depclean/DepCleanGradleAction.java

```diff
@@ -0,0 +1,506 @@
+package se.kth.depclean;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.gradle.api.Action;
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.ConfigurationContainer;
+import org.gradle.api.artifacts.ResolvedArtifact;
+import org.gradle.api.artifacts.ResolvedDependency;
+import org.gradle.api.artifacts.UnresolvedDependency;
+import org.gradle.api.logging.Logger;
+import org.jetbrains.annotations.NotNull;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import se.kth.depclean.analysis.DefaultGradleProjectDependencyAnalyzer;
+import se.kth.depclean.analysis.GradleProjectDependencyAnalysis;
+import se.kth.depclean.core.analysis.ProjectDependencyAnalyzerException;
+import se.kth.depclean.util.JarUtils;
+
+/**
+ * Depclean default and only action.
+ */
+public class DepCleanGradleAction implements Action<Project> {
+
+  // To get some clear visible results.
+  private static final String SEPARATOR = "-------------------------------------------------------";
+
+  // To reduce some code(Frequent use)
+  private static final String sep = File.separator;
+
+  public final Map<ResolvedArtifact, String> artifactConfigurationMap = new HashMap<>();
+
+  @SneakyThrows
+  @Override
+  public void execute(@NotNull Project project) {
```

Comment:
This method is too big, could you refactor it and split it into smaller entities (to improve maintainability)?

Reply:
Yes, I will try to split it as more as possible.